lib/presentation/widget/shared_space/shared_space_widget.dart

```diff
@@ -203,6 +196,25 @@ class _SharedSpaceWidgetState extends State<SharedSpaceWidget> {
     );
   }
 
+  Color _bindingCreateNewButtonColor(AppState appState) {
+    if (!appState.functionalityState.isDriveCreationEnable() && !appState.functionalityState.isCreateWorkgroupEnable()) {
+      return AppColor.disableColor;
+    }
+    return AppColor.primaryColor;
+  }
+
+  void _onPressCreateNewButton(BuildContext context, AppState appState) {
+    if (appState.functionalityState.isDriveEnable() && appState.functionalityState.isDriveCreationEnable() &&
+      appState.functionalityState.isCreateWorkgroupEnable()) {
+      sharedSpaceViewModel.openCreateNewSharedSpaceMenu(context, _createNewSharedSpaceActionTiles(context));
+    } else if (appState.functionalityState.isDriveEnable() && appState.functionalityState.isDriveCreationEnable() &&
+      !appState.functionalityState.isCreateWorkgroupEnable()) {
+      sharedSpaceViewModel.openCreateNewDrive(context);
+    } else if (!appState.functionalityState.isDriveCreationEnable() && appState.functionalityState.isCreateWorkgroupEnable()) {
```

Comment:
IMO U are missing the case where Workgroup is disabled

Reply:
1. drive and workgroup are enabled
2. drive enable and workgroup disable
3. drive disable and workgroup enable
4. both are disable -> no action

Reply:
You just checked `isCreateWorkgroupEnable ` enabled. I don't see a check for `isSharedSpaceEnable `

Reply:
ah, TBH, the creation right is not related to the Workgroup enable or not, it is the same for Drive :)

Reply:
If so you just need to check `isDriveCreationEnable` without `isDriveEnable` in this case.

Reply:
please check again

Reply:
Done